server/src/sim/autofit.py

```diff
@@ -10,83 +10,110 @@ def autofit(D, timelimit=None, maxiters=500, simstartyear=2000, simendyear=2015,
     from numpy import mean, array
     from model import model
     from printv import printv
     from ballsd import ballsd
-    from bunch import Bunch as struct
     from utils import findinds
     from copy import deepcopy
     from updatedata import normalizeF, unnormalizeF
     eps = 0.01 # Don't use too small of an epsilon to avoid divide-by-almost zero errors -- this corresponds to 1% which is OK as an absolute error for prevalence
     printv('Running automatic calibration...', 1, verbose)
-    origM = deepcopy(D.M)
-    origG = deepcopy(D.G)
+    origM = deepcopy(D['M'])
+    origG = deepcopy(D['G'])
     
     # Set options to update year range
     from setoptions import setoptions
-    D.opt = setoptions(D.opt, simstartyear=simstartyear, simendyear=simendyear)
+    D['opt'] = setoptions(D['opt'], simstartyear=simstartyear, simendyear=simendyear)
+    
+    #pdb.set_trace()    
     
     def errorcalc(Flist):
         """ Calculate the error between the model and the data """
         
         printv(Flist, 4, verbose)
         
-        F = list2dict(D.F[0], Flist)
+        F = list2dict(D['F'][0], Flist)
         F = unnormalizeF(F, origM, origG) # CK: Convert from normalized to unnormalized F (NB, Madhura)
-        S = model(D.G, D.M, F, D.opt, verbose=verbose)
-        
-        # Pull out diagnoses data
-        dx = [struct()]
-        dx[0].data = struct()
-        dx[0].model = struct()
-        dx[0].data.x, dx[0].data.y = extractdata(D.G.datayears, D.data.opt.numdiag[0])
-        dx[0].model.x = S.tvec
-        dx[0].model.y = S.dx.sum(axis=0)
+        S = model(D['G'], D['M'], F, D['opt'], verbose=verbose)
         
-        # Prevalence data
-        prev = [struct() for p in range(D.G.npops)]
-        for p in range(D.G.npops): 
-            prev[p].data = struct()
-            prev[p].model = struct()
-            prev[p].data.x, prev[p].data.y = extractdata(D.G.datayears, D.data.key.hivprev[0][p]) # The first 0 is for "best"
-            prev[p].model.x = S.tvec
-            prev[p].model.y = S.people[1:,p,:].sum(axis=0) / S.people[:,p,:].sum(axis=0) # This is prevalence
+
+        # Pull out Prevalence data
+        prev = [dict() for p in range(D['G']['npops'])]
+        for p in xrange(D['G']['npops']): 
+            prev[p]['data'] = dict()
+            prev[p]['model'] = dict()
+            prev[p]['data']['x'], prev[p]['data']['y'] = extractdata(D['G']['datayears'], D['data']['key']['hivprev'][0][p]) # The first 0 is for "best"
+            prev[p]['model']['x'] = S['tvec']
+            prev[p]['model']['y'] = S['people'][1:,p,:].sum(axis=0) / S['people'][:,p,:].sum(axis=0) # This is prevalence
+                                                        
+        #indicators = struct()
+#        indicatorkeys = ['death', 'newtreat', 'numtest','numinfect', 'dx']
+#        #datakeys = ['death', 'newtreat', 'numtest','numinfect', 'dx']
+#        for key in indicatorkeys:
+#            indicators[key] = dict()
```

Comment:
@hugobowne is it intended that every one of those indicators is an array of dictionaries?